packages/runtime/src/runtime/modal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Modal } from "antd";
+import type { NavigateModalScreenAction } from "framework";
+import { createContext, useState } from "react";
+import { createPortal } from "react-dom";
+import { Outlet } from "react-router-dom";
+import OpenInFullIcon from "@mui/icons-material/OpenInFull";
+import CloseFullscreenIcon from "@mui/icons-material/CloseFullscreen";
+
+type ModalProps = {
+  title?: string | React.ReactNode;
+} & Omit<Exclude<NavigateModalScreenAction, string>, "name">;
```

Comment:
Just put the property here explicitly, there's no value to tie it to the action property since it's React code.